pkg/jiraservicedesk/client/client_project.go

```diff
@@ -152,13 +161,41 @@ func (c *jiraServiceDeskClient) DeleteProject(id string) error {
 		return err
 	}
 
-	if response.StatusCode != 204 {
+	if response.StatusCode < 200 || response.StatusCode > 299 {
 		return errors.New("Rest request to delete Project failed with status: " + strconv.Itoa(response.StatusCode))
 	}
 
 	return err
 }
 
+func (c *jiraServiceDeskClient) UpdateProjectAccessPermissions(status bool, key string) error {
```

Comment:
This is being handled in customer update workflow right ?

Reply:
No, it isn't. I think the name caused the confusion. Updated the name.

Reply:
But it should be, what if a user updates the customer CR ?

Reply:
Mistakenly wrote `customer` here a few minutes ago